jdbc/src/java/org/apache/hive/jdbc/HiveStatement.java

```diff
@@ -130,22 +126,20 @@ public class HiveStatement implements java.sql.Statement {
 
   public HiveStatement(HiveConnection connection, TCLIService.Iface client,
       TSessionHandle sessHandle) {
-    this(connection, client, sessHandle, false, 0, DEFAULT_FETCH_SIZE);
+    this(connection, client, sessHandle, false, connection.fetchSize);
   }
 
   public HiveStatement(HiveConnection connection, TCLIService.Iface client, TSessionHandle sessHandle,
-      boolean isScrollableResultset, int initFetchSize, int defaultFetchSize) {
+      boolean isScrollableResultset, int fetchSize) {
```

Comment:
Does this change cause backward compatibility issues?

Reply:
It shouldn't as I know, the original constructor is introduced since Hive 4.0, the HiveConnection and HiveStatement are in a single jar, so they are supposed to be loaded together by the same class loader, I'm not sure how the old `HiveConnection` can point to the new `HiveStatement` in the runtime.